packages/shorebird_code_push_client/lib/src/code_push_client.dart

```diff
@@ -77,6 +79,107 @@ class _CodePushHttpClient extends http.BaseClient {
   }
 }
 
+/// Based on the [IOClient] from the `http` package. The primary difference is
+/// that this implementation uses `stream.addStream` instead of `stream.pipe`
+/// to allow for progress reporting.
+class _UploadProgressHttpClient extends http.BaseClient {
```

Comment:
Rather than extending `BaseClient` couldn't we extend `MultipartRequest` and override `finalize` and expose an `onProgress` callback? It seems like it would be a lot simpler and less invasive.

Reply:
Agreed that it's simpler, but that was the first thing I tried, and the stream progress very quickly completes (goes from 0% to 100% with a bunch of intermediate values) and then hangs while the actual upload happens. I believe the progress we see in the `MultipartRequest` subclass is the file being read, not the upload happening.

Reply:
For posterity, the way I'm testing this is:

1. `shorebird release android`
2. At the release confirmation prompt, turn on the network link condition. I'm using LTE speed.
3. Type 'y'
4. Observe